### ISSUE TITLE
The test case failed using FluentAssertions or Playwright are shown as broken instead of failed

### ISSUE THREAD

**Issue by I37937** [allure-framework/allure-csharp#520]
[//]: # (
. Note: for support questions, please create a discussion at https://github.com/orgs/allure-framework/discussions.
. This repository's issues are reserved for feature requests and bug reports.
.
. Make sure you have a clear name for your issue. The name should start with a capital 
. letter and no dot is required in the end of the sentence. An example of good issue names:
.
. - The report is broken in IE11
. - Add an ability to disable default plugins
. - Support emoji in test descriptions
)

#### I'm submitting a ... 
  - [X] bug report
  - [ ] feature request
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

#### What is the current behavior?
The test case failed using fluentassertions or Playwright are shown as broken instead of failed

#### If the current behavior is a bug, please provide the steps to reproduce and, if possible, a minimal demo of the problem
Create a simple test using FluentAssertions or Playwright dotnet, when you run the tests the resutl is shown as broken
![image](https://github.com/allure-framework/allure-csharp/assets/162322781/65d64dba-6041-4c8e-8c6f-8c808156ae2d)

#### What is the expected behavior?
The test should be failed if I have configured the list of exceptions (failExceptions).

#### Please tell us about your environment:

- OS					 Windows
- .NET framework         .NET 8
- Test framework:        nunit@4.0.0
- Allure adapter:        allure-nunit@2.12.0

#### Other information 

 I was taking a look to the AllureNUnitHelper and I was able to see that the IsBroken function is waiting elements in result.Assertions list but when you use FluentAssertions  or Playwright this list is not update so the tests are always mark as broken.
![image](https://github.com/allure-framework/allure-csharp/assets/162322781/502e443e-f0e5-4aed-9447-d1a649b62f74)

**Comment by gamerka** [comment]
Did you check this page?
https://allurereport.org/docs/nunit-configuration/#allurefailexceptions

**Comment by I37937** [comment]
Yes, I do, I have added the exceptions but I still having the same issue
![image](https://github.com/allure-framework/allure-csharp/assets/162322781/69c35917-b882-4a5c-b180-b69112029bac)

**Comment by arman-avetisyan** [comment]
Hello, I have the same issue,
I am using `Xunit`, `Allure.Net.Commons` and `Shouldly`
When I run a test with this simple step, and generate an Allure report
the test status displayed as Broken
![image](https://github.com/allure-framework/allure-csharp/assets/66781057/732ac7df-a909-4cf1-b63c-b0da85c68acc)

Also, I tried to add the `Shouldly.ShouldAssertException` to the allureConfig.json, the step becomes Failed but the test is still displayed as Broken
![image](https://github.com/allure-framework/allure-csharp/assets/66781057/0b310207-f01d-4b7e-b6ce-2b81e3c65808)

**Comment by Melkor972** [comment]
The same issue with Nunit & FluentAssertions
It seems that the exception type is not tracked by Nunit. It has a stacktrace in a message but doesn't hold an exception type directly. IMO, it would be convenient to detect Failed/Broken status by Nunit's ResultState.Failure/Error respectively

**Comment by ohlori** [comment]
Any chance there's a timeline for this?

**Comment by Dimitri-M-Dumont** [comment]
Patiently waiting on a fix for this too. 😅
